### PR TITLE
feat: OSCO transformer support for OpenScap 1.3.5

### LINKED ISSUE #875: OSCO transformer update to handle OpenScap 1.35.
## Issue description / feature objectives

OSCO version 0.1.44, with OpenScap 1.3.5 has host name in different location than OSCO version 0.1.42, with OpenScap 1.3.3.

## Caveats / Assumptions

In 1.3.3 we get host name from \<target\>.
In 1.3.5 we need to get host name from \<target-facts\>\<fact name="urn:xccdf:fact:asset:identifier:host_name" type="string"\>

## Completion Criteria

OSCO transformer can handle both types of host name specification.

### PR DESCRIPTION
**Pull request by degenaro**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary

Support for OpenScap 1.3.5, in addition to 1.3.3.

Closes #875.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

### CODE REVIEW

File: trestle/transforms/utils/osco_helper.py [hunk 1]
```diff
@@ -61,36 +61,32 @@
...
-        scanner_name: str,
-        scanner_version: str,
-        idref: str,
-        version: str,
-        time: str,
-        result: str,
-        severity: str,
-        weight: str
-    ) -> None:
+    def __init__(self, **args) -> None:
```
Comment (butler54) [inline trestle/transforms/utils/osco_helper.py:64]: I can see what you have done here to circumvent sonar complaining about too many arguments (I think it's limit is 7?).

However, this is pretty ugly as all the arguments are still required and there is now no longer an explicit spec. 
One idea would be to pass these as a dict (after all there will be no overhead as that is what the function call is doing anyway here).

TL;DR: It looks strange.

File: trestle/transforms/utils/osco_helper.py [hunk 2]
```diff
@@ -61,36 +61,32 @@
...
+        self.severity = args['severity']
+        self.weight = args['weight']
+
+    @property
+    def inventory_key(self):
+        """Derive inventory key."""
+        if self.host_name is None:
+            # OpenScap 1.3.3
+            return self.target + ':' + self.target_type
+        else:
```
Comment (butler54) [inline trestle/transforms/utils/osco_helper.py:87]: You don't need the else here as you have returned.

File: trestle/transforms/utils/osco_helper.py [hunk 3]
```diff
@@ -217,19 +224,20 @@ def _parse_xml(self) -> Iterator[RuleUse]:
...
-                    benchmark_id,
-                    scanner_name,
-                    scanner_version,
-                    idref,
-                    version,
-                    time,
-                    result,
-                    severity,
-                    weight
+                    id_=id_,
```
Comment (butler54) [inline trestle/transforms/utils/osco_helper.py:227]: See the comments above. I think a dict might actually make it clearer as the assignment and retrieval can happen at once.

**Review by butler54 — CHANGES_REQUESTED** [review]
See the changes suggested. I'm happy to discuss as they do not change the logic.
One additional point:
It would be great if we can move away from doing filecmp with target files and instead can do attribute comparisons etc. If I change the json output formatting slightly it breaks the test (I can live with what we have right now in the tests).

From a Logic POV looks good to me.

File: trestle/transforms/utils/osco_helper.py (hunk 1 above)
Comment (degenaro) [reply trestle/transforms/utils/osco_helper.py]: I think is is 12 or 13, I missed by 1. Fixed.

File: trestle/transforms/utils/osco_helper.py (hunk 2 above)
Comment (degenaro) [reply trestle/transforms/utils/osco_helper.py:87]: Fixed.

File: trestle/transforms/utils/osco_helper.py (hunk 3 above)
Comment (degenaro) [reply trestle/transforms/utils/osco_helper.py]: Fixed.

**Review by butler54 — APPROVED** [review]
LGTM
